tests/unit/test_args.py

```diff
@@ -459,3 +459,226 @@ def test_client_config_beats_config_store(self):
             client_config=Config(retries={'mode': 'standard'})
         )['client_config']
         self.assertEqual(config.retries['mode'], 'standard')
+
+    def test_does_set_global_sts_endpoint_if_in_legacy_region(self):
+        should = self.args_create._should_set_global_sts_endpoint(
+            region_name='us-west-1', endpoint_url=None, endpoint_config=None
+        )
+        self.assertTrue(should)
```

Comment:
Typically we won't test private methods directly, instead verifying the desired behavior at whatever the top level interface is. That allows us to make more substantial signature changes without having to update existing tests.

How are you intending to retrieve this from the args module? We may need to expose a "public" method even if it's not intended for use, similar to `compute_s3_config`. We can note support expectations in the docstring.

Reply:
> How are you intending to retrieve this from the args module?

`_should_set_global_sts_endpoint` is a method in the class `ClientArgsCreator`. The assignment of builtin parameter values happens within the same class. 

This is, in fact, the reason why the `_compute_endpoint_resolver_v2_builtin_defaults` ended up in `ClientArgsCreator`: Most of the required information is available locally in this class.